pytest_deep_analysis/checkers.py

```
@@ -425,100 +601,198 @@ def close(self) -> None:
+        complexity_indicators = {
+            "connection", "cursor", "execute", "commit", "rollback",
+            "session", "transaction", "database", "db",
+            "request", "response", "http", "api"
+        }
+
+        code_str = node.as_string().lower()
+        if any(indicator in code_str for indicator in complexity_indicators):
+            return True
+
+        # Check for yield (resource management fixtures)
+        for yield_node in node.nodes_of_class(astroid.Yield):
```

Ruff (0.14.4)

Loop control variable `yield_node` not used within loop body

Rename unused `yield_node` to `_yield_node`

(B007)